ecs-agent/tmds/handlers/fault/v1/handlers/handlers.go

```diff
@@ -630,17 +701,40 @@ func (h *FaultHandler) CheckNetworkLatency() func(http.ResponseWriter, *http.Req
 		rwMu.RLock()
 		defer rwMu.RUnlock()
 
-		// TODO: Check status of current fault injection
-		// TODO: Return the correct status state
-		responseBody := types.NewNetworkFaultInjectionSuccessResponse("running")
-		logger.Info("Successfully checked status for fault", logger.Fields{
+		// Check and return the status of current fault injection.
+		var responseBody types.NetworkFaultInjectionResponse
+		var httpStatusCode int
+		stringToBeLogged := "Failed to check status for fault"
+		// All command executions for the check network latency workflow all together should finish within 5 seconds.
+		// Thus, create the context here so that it can be shared by all os/exec calls.
+		ctx, cancel := h.osExecWrapper.NewExecContextWithTimeout(context.Background(), requestTimeoutDuration)
+		defer cancel()
+		// Check the status of current fault injection.
+		latencyFaultExists, _, err := h.checkTCFault(ctx, taskMetadata)
+		if errors.Is(ctx.Err(), context.DeadlineExceeded) {
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(fmt.Sprintf(requestTimedOutError, requestType))
+			httpStatusCode = http.StatusInternalServerError
+		} else if err != nil {
+			responseBody = types.NewNetworkFaultInjectionErrorResponse(err.Error())
+			httpStatusCode = http.StatusInternalServerError
+		} else {
+			// If there already exists a fault in the task network namespace.
+			if latencyFaultExists {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("running")
+				httpStatusCode = http.StatusOK
+			} else {
+				responseBody = types.NewNetworkFaultInjectionSuccessResponse("not-running")
+				httpStatusCode = http.StatusOK
+			}
```

Comment:
Q: should we be logging that we've successfully checked the status of the fault here?

Reply:
Yes. Thanks for the catch! Will address this in the next PR.